pkg/exporter/sumologicexporter/translate_metrics_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+func TestTranslateTelegrafMetric_NamesAreTranslatedCorrectly(t *testing.T) {
+	testcases := []struct {
+		nameIn  string
+		nameOut string
+	}{
+		// CPU metrics
+		{
+			nameIn:  "cpu_usage_active",
+			nameOut: "CPU_Total",
+		},
+		{
+			nameIn:  "cpu_usage_idle",
+			nameOut: "CPU_Idle",
+		},
+		{
+			nameIn:  "cpu_usage_iowait",
+			nameOut: "CPU_IOWait",
+		},
+		{
+			nameIn:  "cpu_usage_irq",
+			nameOut: "CPU_Irq",
+		},
+		{
+			nameIn:  "cpu_usage_nice",
+			nameOut: "CPU_Nice",
+		},
+		{
+			nameIn:  "cpu_usage_softirq",
+			nameOut: "CPU_SoftIrq",
+		},
+		{
+			nameIn:  "cpu_usage_steal",
+			nameOut: "CPU_Stolen",
+		},
+		{
+			nameIn:  "cpu_usage_System",
+			nameOut: "CPU_Sys",
+		},
+		{
+			nameIn:  "cpu_usage_user",
+			nameOut: "CPU_User",
+		},
+		{
+			nameIn:  "system_load1",
+			nameOut: "CPU_LoadAvg_1min",
+		},
+		{
+			nameIn:  "system_load5",
+			nameOut: "CPU_LoadAvg_5min",
+		},
+		{
+			nameIn:  "system_load15",
+			nameOut: "CPU_LoadAvg_15min",
+		},
+		// Disk metrics
+		{
+			nameIn:  "disk_used",
+			nameOut: "Disk_Used",
+		},
+		{
+			nameIn:  "disk_used_percent",
+			nameOut: "Disk_UsedPercent",
+		},
+		{
+			nameIn:  "disk_inodes_free",
+			nameOut: "Disk_InodesAvailable",
+		},
+		// Disk IO metrics
+		{
+			nameIn:  "diskio_reads",
+			nameOut: "Disk_Reads",
+		},
+		{
+			nameIn:  "diskio_read_bytes",
+			nameOut: "Disk_ReadBytes",
+		},
+		{
+			nameIn:  "diskio_writes",
+			nameOut: "Disk_Writes",
+		},
+		{
+			nameIn:  "diskio_write_bytes",
+			nameOut: "Disk_WriteBytes",
+		},
+
+		// Memory metrics
+		{
+			nameIn:  "mem_total",
+			nameOut: "Mem_Total",
+		},
+		{
+			nameIn:  "mem_free",
+			nameOut: "Mem_free",
+		},
+		{
+			nameIn:  "mem_available",
+			nameOut: "Mem_ActualFree",
+		},
+		{
+			nameIn:  "mem_used",
+			nameOut: "Mam_ActualUsed",
+		},
+		// TODO rethink above and below
+		// {
+		// 	nameIn:  "mem_used",
+		// 	nameOut: "Mam_Used",
+		// },
```

Comment:
Please create github issue for that

Reply:
Done: https://github.com/SumoLogic/sumologic-otel-collector/issues/112